src/Numerics/Mesh/Filters.jl

```diff
@@ -257,27 +367,52 @@ horizontal and/or vertical reference directions.
             filterinξ3 = dim == 2 ? false : true
         end
 
-        nfilterstates = length(states)
+        nstates = number_filter_state(target, FT)
+        nfilterstates = number_filter_filtered(target, FT)
+        nfilteraux = number_filter_auxiliary(target, FT)
+
+        # ugly workaround around problems with @private
+        # hopefully will be soon fixed in KA
+        l_Q2 = MVector{nstates, FT}(undef)
+        l_Qfiltered2 = MVector{nfilterstates, FT}(undef)
     end
 
     s_filter = @localmem FT (Nq, Nq)
     s_Q = @localmem FT (Nq, Nq, Nqk, nfilterstates)
+    l_Q = @private FT (nstates,)
     l_Qfiltered = @private FT (nfilterstates,)
+    l_aux = @private FT (nfilteraux,)
 
     e = @index(Group, Linear)
     i, j, k = @index(Local, NTuple)
 
     @inbounds begin
+        ijk = i + Nq * ((j - 1) + Nq * (k - 1))
+
         s_filter[i, j] = filtermatrix[i, j]
 
+        @unroll for s in 1:nstates
+            l_Q[s] = Q[ijk, s, e]
+        end
```

Comment:
Is there a reason we need to load the whole state and not just the elements we need?
```julia
        @unroll for (s,is) in enumerate(I)
            l_Q[s] = Q[ijk, is, e]
        end
```

Same for auxiliary below.

Should we warn somewhere the assumption is the `state` and `filtered_state` match in order? (Same with auxiliary).

That is we cannot do something like:
```julia
function vars_state_consvative(::MyModel, FT)
  @vars begin
    a::FT
    b::FT
    c::FT
  end
end
function vars_filtered_state(::MyModel, FT)
  @vars begin
    c::FT
  end
end
FilterIndices(3)
```

Since if I am not mistake, the numbers of filtered states would be 3 in this case.

Reply:
I am not sure if the compiler is optimizing out these loads or not.  If not and it is straight forward to do so, it would be nice to minimize the loads.